torch/csrc/dynamo/eval_frame.c

```diff
@@ -1010,7 +1023,8 @@ static struct PyModuleDef _module = {
     _methods};
 
 PyObject* torch_c_dynamo_eval_frame_init(void) {
-  extra_index = _PyEval_RequestCodeExtraIndex(ignored);
+  cache_entry_extra_index = _PyEval_RequestCodeExtraIndex(ignored);
+  dynamic_plan_extra_index = _PyEval_RequestCodeExtraIndex(ignored);
```

Comment:
Why did you decide to allocate two extra indices, as opposed to storing it all on the cache entry?

Reply:
The cache entry is an element in a linked list. There are many per frame, and they chain each other. I don't want to duplicate the plan across all of them. There is 1 plan per frame.

Reply:
At the cost of one more pointer chase for cache entry, you could just make a structure that holds the linked list of cache entries, as well as the frame state. This also seems like a better structure if we end up setting a tree guard structure, since you would probably just store the dispatch continuation as a function on this structure (removing the pointer chase). You would also be able to pass this structure directly to Python for introspection.